bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/product/ProductImpl.java

```diff
@@ -387,4 +387,40 @@ private String safeDescription(ProductInterface product) {
         return xssApi.filterHTML(description.getHtml());
     }
 
+    // DataLayer methods
+
+    @Override
+    public ComponentData getComponentData() {
+        return new ProductDataImpl(this, resource);
+    }
+
+    @Override
+    protected String generateId() {
+        return StringUtils.join("product", ID_SEPARATOR, productRetriever.fetchProduct().getId());
```

Comment:
Why do you use the `slug` for the productlistitem, but the `id` for the product component?

Reply:
I think we should completely avoid using `id` because in Magento this is basically the row id of the database entry, so nothing reusable or meaningful. Better use the `sku` everywhere, because the `slug` can change over time while the `sku` should never change.